### ISSUE TITLE
CA2249 (prefer Contains instead of IndexOf) does not trigger when IndexOf comparison operator is `!=`

### ISSUE THREAD

**Issue by carlossanlop** [dotnet/roslyn-analyzers#6587]
### Analyzer

**Diagnostic ID**: [CA2249](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca2249): `Consider using String.Contains instead of String.IndexOf`

### Analyzer source

The one currently being consumed in dotnet/runtime:

<details><summary>dotnet --info (expand)</summary>

```
$ c:\repos\runtime\.dotnet\dotnet --info (expand)

.NET SDK:
 Version:   8.0.100-preview.1.23115.2
 Commit:    913e4679b3

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22621
 OS Platform: Windows
 RID:         win10-arm64
 Base Path:   c:\repos\runtime\.dotnet\sdk\8.0.100-preview.1.23115.2\

Host:
  Version:      8.0.0-preview.1.23110.8
  Architecture: arm64
  Commit:       7deac7d6da

.NET SDKs installed:
  8.0.100-preview.1.23115.2 [c:\repos\runtime\.dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 8.0.0-preview.1.23112.2 [c:\repos\runtime\.dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 8.0.0-preview.1.23110.8 [c:\repos\runtime\.dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 8.0.0-preview.1.23112.2 [c:\repos\runtime\.dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  x64   [C:\Program Files\dotnet\x64]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x64\InstallLocation]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  C:\repos\runtime\global.json

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```
</details>

### Describe the bug

The rule is only getting triggered when `IndexOf` is compared with `==` but not with `!=`.

### Steps To Reproduce

I am currently reproducing it in this PR: https://github.com/dotnet/runtime/pull/82812#discussion_r1167071021

### Expected behavior

Only works for `==`:

<img src="https://user-images.githubusercontent.com/1175054/232145026-5ff8b83d-e542-493c-9cc2-41af6dba766d.png" width="400"/>

### Actual behavior

Does not work for `!=`:

<img src="https://user-images.githubusercontent.com/1175054/232144899-5c187eb0-f7d9-4b60-8c89-3a726c58a8c1.png" width="400"/>

## Additional context

The analyzer code that verifies the comparison operators being used is only considering `==` and `>=` but not `!=`:

https://github.com/dotnet/roslyn-analyzers/blob/432aaef704712d79d436a79fafddbd1e9c4409ea/src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfAnalyzer.cs#L178-L179

We enable rule CA2249 as warning in the dotnet/runtime repo:
https://github.com/dotnet/runtime/blob/9c02b25f08c8f1aaf7811446325ac9f873efa451/eng/CodeAnalysis.src.globalconfig#L622-L623

Related issues:
- https://github.com/dotnet/roslyn-analyzers/issues/3655
- https://github.com/dotnet/roslyn-analyzers/issues/3657

Related PRs:
- https://github.com/dotnet/runtime/issues/33785 (closed)
- https://github.com/dotnet/roslyn-analyzers/pull/3610 (closed)
